manager/src/main/java/org/openremote/manager/mqtt/ActiveMQORSecurityManager.java

```diff
@@ -98,13 +104,58 @@ protected Subject getAuthenticatedSubject(String user,
         String realm = null;
         ClassLoader currentLoader = Thread.currentThread().getContextClassLoader();
         ClassLoader thisLoader = this.getClass().getClassLoader();
-
+        X509Certificate[] certs = getCertsFromConnection(remotingConnection);
         if (user != null) {
             String[] realmAndUsername = user.split(":");
             if (realmAndUsername.length == 2) {
                 realm = realmAndUsername[0];
                 user = realmAndUsername[1];
             }
+        }else if (certs != null && certs.length > 0) {
+
+            X509Certificate leaf = certs[0];
+            String dn = leaf.getSubjectX500Principal().getName();
+            try {
+                // Check for Client Authentication EKU (This denotes that this is the actual client certificate to enforce)
+                if(!leaf.getExtendedKeyUsage().contains("1.3.6.1.5.5.7.3.2")) {
+                    // TODO: Not sure about what extent to which we would like to enforce this.
+                    // I have seen codebases use this EKU key to ensure that the certificate they are examining for client
+                    // auth is the correct one to inspect.I will have to perform some more research about it, but I think
+                    // that the client certificate is always going to be the leaf. For now, log a warning about it.
+                    LOG.log(Level.WARNING, "Presented certificate DOES NOT have Client Authentication Extended Key Usage. " +
+                            "Attempting to use, but please fix this for subsequent requests.");
+                }
+                LdapName ldapName = new LdapName(dn);
+                for (Rdn rdn : ldapName.getRdns()) {
+                    String type = rdn.getType();
+                    String value = rdn.getValue().toString();
+                    if ("OU".equalsIgnoreCase(type) && realm == null) {
```

Comment:
put `realm == null` check first it is much quicker than the string comparison check (don't think the compiler is smart enough to reorder such things for you)